lib/screens/create_region_event_screens/chose_event_date_and_time.dart

```diff
@@ -0,0 +1,94 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:intl/intl.dart';
+import 'package:seeds/components/flat_button_long.dart';
+import 'package:seeds/domain-shared/ui_constants.dart';
+import 'package:seeds/screens/create_region_event_screens/components/date_time_widget_row.dart';
+import 'package:seeds/screens/create_region_event_screens/interactor/viewmodels/create_region_event_bloc.dart';
+import 'package:seeds/screens/create_region_event_screens/interactor/viewmodels/create_region_events_page_commands.dart';
+
+class ChoseEventDateAndTime extends StatelessWidget {
+  const ChoseEventDateAndTime({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocConsumer<CreateRegionEventBloc, CreateRegionEventState>(
+      listenWhen: (_, current) => current.pageCommand != null,
+      listener: (context, state) async {
+        final pageCommand = state.pageCommand;
+
+        if (pageCommand is ShowPickDate) {
+          final selected = await showDatePicker(
+            context: context,
+            initialDate: DateTime.now(),
+            firstDate: DateTime.now(),
+            lastDate: DateTime(2099),
+          );
+          if (selected != null) {
+            // ignore: use_build_context_synchronously
+            BlocProvider.of<CreateRegionEventBloc>(context).add(OnSelectDateChanged(selected));
+          }
+        } else if (pageCommand is ShowPickTime) {
+          final TimeOfDay? picked = await showTimePicker(
+            context: context,
+            initialTime: const TimeOfDay(hour: 00, minute: 00),
+          );
+          if (picked != null) {
```

Comment:
gaby, i know i said to check for null. 
But why dont we send a nullable to the event? And move this null check logic over to the bloc?